apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressBalancesAlph.ts

```diff
@@ -21,9 +21,10 @@ import { useQuery } from '@tanstack/react-query'
 import { UseFetchAddressProps } from '@/api/apiDataHooks/address/addressApiDataHooksTypes'
 import { addressAlphBalancesQuery } from '@/api/queries/addressQueries'
 import { useAppSelector } from '@/hooks/redux'
+import { selectCurrentlyOnlineNetworkId } from '@/storage/settings/networkSelectors'
 
 const useFetchAddressBalancesAlph = ({ addressHash, skip }: UseFetchAddressProps) => {
-  const networkId = useAppSelector((s) => s.network.settings.networkId)
+  const networkId = useAppSelector(selectCurrentlyOnlineNetworkId)
```

Comment:
this was hard to find out, but essentially, when switching networks the Redux state was being updated BEFORE the `throttledClient` was updated, leading to weird states (that were preserved in indexedDB) where `mainnet` endpoints had `testnet` data.